model/src/main/java/org/projectnessie/api/TreeApi.java

```diff
@@ -91,7 +91,19 @@
   /** Create a new reference. */
   @POST
   @Path("tree")
-  @Operation(summary = "Create a new reference")
+  @Operation(
+      summary = "Create a new reference",
```

Comment:
is this for authz? We can't create a branch from a hash we can't see? If yes can you make it clear thats the reason. It looks like a confusing api.

Reply:
right - the whole PR is for proper authz.

Reply:
There are two options IMO:
1. use the approach in this PR (i.e. just add that `sourceRefName`, which is confusing, right) and keep it like that
2. in this PR: also refactor the `Branch`, `Tag`, `Hash`, `Reference` types -> split into type-hierarchy like `NamedRef` (so type-name only) plus a separate one with type+name+hash
3. option 1 + option 2 in a follow-up